File: mixin-base/src/main/java/net/yakclient/mixin/base/internal/instruction/core/CoreInsnExecutor.java
package net.yakclient.mixin.base.internal.instruction.core;

import net.yakclient.mixin.base.internal.instruction.Instruction;
import net.yakclient.mixin.base.internal.instruction.InstructionExecutor;
import org.objectweb.asm.MethodVisitor;

public class CoreInsnExecutor implements InstructionExecutor {
    private final Instruction instruction;
    private final MethodVisitor visitor;

    public CoreInsnExecutor(Instruction instruction, MethodVisitor visitor) {
        this.instruction = instruction;
        this.visitor = visitor;
    }

    @Override
    public void execute() {
        instruction.getInstructions().accept(visitor);
//        for (AbstractInsnNode insn : instruction.getInstructions()) {
//            insn.accept(visitor);
//        }
    }

}

File: mixin-base/src/main/kotlin/net/yakclient/mixin/base/internal/instruction/core/CoreInsnExecutor.kt
package net.yakclient.mixin.base.internal.instruction.core

import net.yakclient.mixin.base.internal.instruction.Instruction
import net.yakclient.mixin.base.internal.instruction.InstructionExecutor
import org.objectweb.asm.MethodVisitor

class CoreInsnExecutor(private val instruction: Instruction, private val visitor: MethodVisitor) : InstructionExecutor {
    override fun execute() {
        instruction.insn.accept(visitor)
    }
}